### PR TITLE
Advanced Mapping: add field mappings cascade delete

### PR DESCRIPTION
**Pull request by psadie-sfdo**
- "Deleting" an object mapping will now "delete" all of its field mapping children.

# Critical Changes

# Changes
## NPSP Data Import Enhancements

- We added cascading delete to Advanced Mapping for Field Mappings. Now, when a user deletes an Object Mapping, all of the Field Mappings within the Object Mapping are deleted.

# Issues Closed

# New Metadata

# Deleted Metadata

### CODE REVIEW

**Comment by psadie-sfdo** [comment]
**lurch: attach W-035505

**Comment by LurchTheButler** [comment]
Tracking <a href="https://foundation.my.salesforce.com/a2x1E000001C9obQAC" target="_blank">W-035505</a>

**Comment by psadie-sfdo** [comment]
**lurch: attach W-035505

File: src/classes/BDI_ManageAdvancedMappingCtrl.cls [hunk 1]
```diff
@@ -380,6 +432,7 @@ public class BDI_ManageAdvancedMappingCtrl {
             for(String fieldName : sObjectCastedCustomMetadata.getPopulatedFieldsAsMap().keySet()) {
 
                 Boolean invalidField =
+                    fieldName.contains('__r') ||
```
Comment (ptewson-sfdo) [inline src/classes/BDI_ManageAdvancedMappingCtrl.cls]: What is exactly is causing us to block the relationship fields - was that not an issue before we started handling cascade delete, or is this a ride-along fix for a separate issue?

Comment (psadie-sfdo) [reply src/classes/BDI_ManageAdvancedMappingCtrl.cls]: Ah, neither actually. This is an artifact from when I was trying out another way of cascade deleting. Pushed a new commit reverting this. Thank you!

**Pull request merged**